### PR TITLE
Implementation of annotation and timeline client-side services.

### PR DESCRIPTION
**Pull request by olaStream**
Adding the implementation to for the new client-side services - Timeline and Annotation

### CODE REVIEW

**Comment by andreas-marschke** [comment]
Awesome work Ola!

Can you split out the 2 Keys from Repository.js into `Repository.js` and `Annotation.js` and `Timeline.js`? As far as I can see none of them share any actual functionality. You can then create a new SOASTA.js which will require the 3 previous JS files. 

ie.:

```
SOASTA = {
  Repository: require("./Repository.js"),
  Timeline: require("./Timeline.js"),
  Annotation: require("./Annotation.js")
};
```

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "soasta-repository",
-  "version": "2.0.2",
+  "version": "2.0.3",
```
Comment (nicjansma) [inline package.json]: I'd probably suggest this is at least a minor point release - `2.1.0`?

File: lib/model/Annotation.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+ * Find all Annotation matching {@link query} of a {@link type} in the repository
+ *
+ * @param {string} token - API-Auth token retrieved with {@link Tokens}
+ * @param {(Object|function|String[])} query - Filter attributes to apply to repository search
+ * @param {bool} includeDetails - Flag if or if not to return full object from query
+ * @param {function} callback -
+ * [A callback function called after the API request was finalized]{@link Annotation~getAnnotationObjectsList(callback)}
+ */
+Annotation.prototype.getAnnotationObjectsList = function(token, query, callback) {
+  log.debug("queryObjects(token=" + token + ",  query=" + JSON.stringify(query) + ")");
```
Comment (nicjansma) [inline lib/model/Annotation.js]: 2-spaces here?

**Comment by nicjansma** [comment]
Can you update `Gruntfile.js` to update `lintFiles[]` and the `jsdoc` task (looks like it should use `jsdocSourceFiles` in the task definition too)?

Should we add the two new CLIs to `index.js`?

Updates to `README.md` for `cmdAnnotation.js` and `cmdTimeline.js`?

Do we want to add `bins` in `package.json` for the new `cmd*`?

```
"bin": {
    "soasta-repository": "./cmd.js",
    "soasta-repository-timeline": "./cmdTimeline.js",
    "soasta-repository-annotation": "./cmdAnnotation.js",
  },
```

Are there any tests we can add to validate?

**Pull request merged**